lib/improver/tests/ensemble_copula_coupling/ensemble_copula_coupling/test_GeneratePercentilesFromProbabilities.py

```diff
@@ -492,6 +541,20 @@ def test_check_data_not_specifying_percentiles(self):
         result = plugin.process(cube)
         self.assertArrayAlmostEqual(result.data, data)
 
+    def test_check_data_over_specifying_percentiles(self):
+        """
+        Test that the plugin raises a suitable error when both a number and set
+        or percentiles are specified.
+        """
+        no_of_percentiles = 3
+        percentiles = [25, 50, 75]
```

Comment:
Need a unit test with single percentile

Reply:
Added a version with only the 25th

Reply:
And I added one where it is not defined as a list at all to ensure that it can deal with a single value specified as a float.